library/HTMLPurifier/AttrDef/CSS/Gradient.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/**
+ * Validates Gradient as defined by CSS.
+ * Works with
+ */
+class HTMLPurifier_AttrDef_CSS_Gradient extends HTMLPurifier_AttrDef
+{
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Color
+     */
+    protected $color;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Direction
+     */
+    protected $direction;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Angle
+     */
+    protected $angle;
+
+    /**
+     * @type HTMLPurifier_AttrDef_CSS_Composite
+     */
+    protected $size;
+
+    public function __construct()
+    {
+        $this->color = new HTMLPurifier_AttrDef_CSS_Color();
+        $this->direction = new HTMLPurifier_AttrDef_CSS_Direction();
+        $this->angle = new HTMLPurifier_AttrDef_CSS_Angle();
+        $this->size = new HTMLPurifier_AttrDef_CSS_Composite(
+            array(
+                new HTMLPurifier_AttrDef_CSS_Length(),
+                new HTMLPurifier_AttrDef_CSS_Percentage()
+            )
+        );
+    }
+
+    /**
+     * @param string $string
+     * @param \HTMLPurifier_Config $config
+     * @param \HTMLPurifier_Context $context
+     * @return bool|string
+     */
+    public function validate($string, $config, $context)
+    {
+        if (strpos($string, 'gradient(') === false) {
+            return false;
+        }
+
+        // Get gradient function (linear, radial, repeating-linear, repeating-radial)
+        $function = explode('gradient', $string);
+        $function = reset($function) . 'gradient';
```

Comment:
I find this line exceptionally strange. Why are you concatenating gradient at the end of function?